### PR TITLE
[feature][Rest-Api] Add get log info api.

### PR DESCRIPTION
**Pull request by liugddx**
### Purpose of this pull request

add get log info api.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

ut case

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by Hisoka-X — COMMENTED** [review]
I don’t quite understand the usage scenario of this PR. cc @EricJoy2048 @hailin0

**Review by davidzollo — DISMISSED** [review]

**Comment by davidzollo** [comment]
> I don’t quite understand the usage scenario of this PR. cc @EricJoy2048 @hailin0

get hazelcast log from api?

**Comment by liugddx** [comment]
like flink web ui.we can get log from rest api

**Comment by liugddx** [comment]
> like flink web ui.we can get log from rest api

Later we can inject the job ID into the log as trace_id. Later, the execution of the task can be tracked through elk or other logging engines.

**Comment by hailin0** [comment]
Please upload running screenshots

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java [hunk 1]
```diff
@@ -44,6 +44,11 @@ public class RestConstant {
     public static final String ERROR_MSG = "errorMsg";
 
     public static final String METRICS = "metrics";
+
+    public static final String LOG_PATH = "seatunnel.logs.path";
+
+    public static final String LOG_FILE_NAME = "seatunnel.logs.file_name";
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java:50]: Please add a comment indicating that this configuration is associated with a shell script

**Comment by Hisoka-X** [comment]
> > like flink web ui.we can get log from rest api
> 
> Later we can inject the job ID into the log as trace_id. Later, the execution of the task can be tracked through elk or other logging engines.

Thanks for explain this. I have some question about get log api.
SeaTunnel may have more than one log file and there will be many archived logs. How should we deal with them?
SeaTunnel have more than one node. Do we currently only support obtaining logs on the master node?

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 2]
```diff
@@ -159,6 +166,19 @@ private void getSystemMonitoringInformation(HttpGetCommand command) {
         this.prepareResponse(command, jsonValues);
     }
 
+    private void handleLog(HttpGetCommand command) throws IOException {
+
+        String logPath = System.getProperty("seatunnel.logs.path");
+        String logName = System.getProperty("seatunnel.logs.file_name") + ".log";
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:174]: Why not use config in ${SEATUNNEL_HOME}/config/log4j2.properties to get file path? Declaring the log path with different keys in different places will increase maintenance costs.

Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:174]: > Why not use config in ${SEATUNNEL_HOME}/config/log4j2.properties to get file path? Declaring the log path with different keys in different places will increase maintenance costs. 

How to get this? Can you elaborate?

**Comment by liugddx** [comment]
> > > like flink web ui.we can get log from rest api
> > 
> > 
> > Later we can inject the job ID into the log as trace_id. Later, the execution of the task can be tracked through elk or other logging engines.
> 
> Thanks for explain this. I have some question about get log api. SeaTunnel may have more than one log file and there will be many archived logs. How should we deal with them? SeaTunnel have more than one node. Do we currently only support obtaining logs on the master node?

1. Just like flink, you can provide an interface /logs to obtain all log files, and then provide detailed information.
2. Yes, currently only master logs can be obtained. I can view the log information of the specified machine based on nodeId, like /{nodeId}/log

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java (hunk 2 above)
Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:174]: oh, seem like we already introduce `seatunnel.logs.path` and `seatunnel.logs.file_name` environment variables before this PR. Ignore this one.

**Comment by Hisoka-X** [comment]
> Yes, currently only master logs can be obtained. I can view the log information of the specified machine based on nodeId, like /{nodeId}/log

If this is TODO, could you create an issue for this? Thanks.

**Comment by liugddx** [comment]
> Please upload running screenshots



> Please upload running screenshots

image

**Comment by liugddx** [comment]
> SeaTunnel may have more than one log file and there will be many archived logs. How should we deal with them?

I misunderstood. Currently, the log information of the specified node is obtained directly through the IP address. Users can directly obtain the log information of the specified node through this request address.

**Comment by EricJoy2048** [comment]
I think we should make a detailed design and discussion on this feature. There are many problems to be solved in obtaining logs, such as how to deal with log switching, how to prevent log reading from affecting the stability of master services, and so on.

**Pull request closed without merging**